luarules/gadgets/cmd_nano_range_check.lua

```diff
@@ -0,0 +1,148 @@
+function gadget:GetInfo()
+	return {
+		name = "Construction Turrets Range Check",
+		desc = "Stops construction turrets from getting assigned to guards, repair, reclaim and attacks out of reach.",
+		author = "Nehroz",
+		date = "2024.12.01",
+		layer = 0,
+		enabled = true,
+	}
+end
+
+if not gadgetHandler:IsSyncedCode() then return end
+Spring.SetLogSectionFilterLevel(gadget:GetInfo().name, LOG.INFO) -- WARN Remove in final version.
+Spring.Log(gadget:GetInfo().name, LOG.INFO,"Unsync Load.")
+
+
+--- key is the nano's ID, followed by a Set-like with the unitID of target currently inside the nano's range.
+--- Used for CommandArray manipulation
+---@type {[number] : {number : boolean}}
+local trackingTable = {}
+
+local constructionTurretsDefs = {}
+for unitDefID, unitDef in ipairs(UnitDefs) do
+	if unitDef.isFactory == false and unitDef.isStaticBuilder then
+		constructionTurretsDefs[unitDefID] = {
+			maxBuildDistance = unitDef.buildDistance + unitDef.radius,
+			buildRange3D = unitDef.buildRange3D,
+		}
+	end
+end
+
+local function isEmptyTable(tbl)
+	for _ in pairs(tbl) do
+		return false
+	end
+	return true
+end
+
+-- only used for Logging; remove with it.
+local function getTableSize(tbl)
+	local i = 0
+	for _ in pairs(tbl) do
+		i = i + 1
+	end
+	return i
+end
+
+function gadget:Initialize()
+	gadgetHandler:RegisterAllowCommand(CMD.REPAIR)
+	gadgetHandler:RegisterAllowCommand(CMD.GUARD)
+	gadgetHandler:RegisterAllowCommand(CMD.RECLAIM)
+end
+
+local gate = false
+function gadget:GameFrame(frame)
+	local removes = 0
+	local inside = 0
+	local nanosRun = 0
+	local unitsCalced = 0
+
+	if frame % 60 > 10 then return end -- run 10 frames, skip 50; allow LuaUser.cpp to catch up. running full every 1s
```

Comment:
The comment is incorrect, 60 frames is 2s not 1s. If you want it to specifically be "however many frames is 1s" (or 2s) then use `Game.gameSpeed`.

Reply:
uh, so the game only a fixed time step of 30, gotcha. Was under the impression that it just runs with 60, but did not look into that.